makefile.shade

```diff
@@ -751,6 +698,62 @@ functions @{
     private static readonly string _packagesSubDir = "packages";
     private static readonly string _runtimesSubDir = "runtimes";
 
+    int ExecuteAndRedirectOutput(string command, string argument, string outputFile)
+    {
+        var procStartInfo = new ProcessStartInfo 
+        {
+            FileName = command, 
+            Arguments = argument, 
+            WorkingDirectory = Directory.GetCurrentDirectory(),
+            UseShellExecute = false
+        };
+        procStartInfo.EnvironmentVariables["KRE_TRACE"] = "0";
+        procStartInfo.RedirectStandardOutput = true;
+
+        var process = Process.Start(procStartInfo);
+        var content = process.StandardOutput.ReadToEnd();
+        process.WaitForExit();
+
+        File.WriteAllText(outputFile, content);
+
+        return process.ExitCode;
+    }
+
+    HashSet<string> ListDependencyAssemblies(string projectFolder, string framework, string coreclrFolder = null)
+    {
+        var tmpFile = Path.GetTempFileName();
+        var argument = string.Format("list {0} --framework {1} --assemblies", projectFolder, framework);
+
+        if (coreclrFolder != null)
+        {
+            argument += (" --runtime " + coreclrFolder);
+        }
+
+        int result;
+        if (!IsMono)
+        {
+            result = ExecuteAndRedirectOutput("cmd", string.Format("/C kpm {0}", argument), tmpFile);
+        }
+        else 
+        {
+            result = ExecuteAndRedirectOutput("kpm", argument, tmpFile);
```

Comment:
Call it ``ExecuteAndCaptureOutput `. What about exit codes? Should it return a tuple?